src/RouteCollector.php

```diff
@@ -27,13 +28,74 @@ public function __construct(RouteParser $routeParser, DataGenerator $dataGenerat
      * @param mixed  $handler
      */
     public function addRoute($httpMethod, $route, $handler) {
+        $route = $this->prependRouteWithRouteGroup($route);
         $routeDatas = $this->routeParser->parse($route);
         foreach ((array) $httpMethod as $method) {
             foreach ($routeDatas as $routeData) {
                 $this->dataGenerator->addRoute($method, $routeData, $handler);
             }
         }
     }
+
+    /**
+     * Prepends the route string with the route string specified in the route group.
+     *
+     * @param string $route
+     * @return string
+     */
+    protected function prependRouteWithRouteGroup($route)
+    {
+        return $this->prependRouteWithGroupsRoute($route, $this->currentRouteGroup);
+    }
+
+
+    /**
+     * Prepends the provided route with the route inside the provided route group.
+     *
+     * @param string $route
+     * @param null|\stdClass $group
+     * @return string
+     */
+    private function prependRouteWithGroupsRoute($route, $group)
+    {
+        if (is_object($group)) {
+            $route = "{$group->route}/" . ltrim($route, '/');
```

Comment:
Right now the implementation trims the $group->route (during initialization), ltrims the $route and manually inserts a slash between them. I'm wondering if we should really do this instead of just concatenating them as-is. This allows a wider range of applications for groups, e.g. instead of the example

```php
$r->addGroup('/admin', function ($r) {
    $r->addRoute('GET', '/do-something', 'handler');
    $r->addRoute('GET', '/do-another-thing', 'handler');
    $r->addRoute('GET', '/do-something-else', 'handler');
});
```

you could also do something like

```php
$r->addGroup('/admin-', function ($r) {
    $r->addRoute('GET', 'do-something', 'handler');
    $r->addRoute('GET', 'do-another-thing', 'handler');
    $r->addRoute('GET', 'do-something-else', 'handler');
});
```

to get `/admin-do-something` URLs. Now, I don't know how practically useful allowing such other formats is...